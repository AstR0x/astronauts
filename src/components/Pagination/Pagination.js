import React from 'react';

import styles from './Pagination.module.css';

export const Pagination = ({ onChangePage, totalPages, currentPage }) => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalPages); i++) {
    pageNumbers.push(i);
  }

  const buttonClasses = pageNumber => {
    const classes = [styles.listButton];

    if (pageNumber === currentPage) {
      classes.push(styles.isActive);
    }

    return classes.join(' ');
  };

  return (
    <ul className={styles.list}>
      <li>
        <button
          onClick={onChangePage.bind(null, 1)}
          className={styles.listButton}
          type="button"
        > &lt;&lt;
        </button>
      </li>
      {pageNumbers.map(pageNumber => (
        <li key={pageNumber}>
          <button
            onClick={onChangePage.bind(null, pageNumber)}
            className={buttonClasses(pageNumber)}
            type="button"
          >
            {pageNumber}
          </button>
        </li>
      ))}
      <li>
        <button
          onClick={onChangePage.bind(null, totalPages)}
          className={styles.listButton}
          type="button"
        >
          &gt;&gt;
        </button>
      </li>
    </ul>
  );
};
