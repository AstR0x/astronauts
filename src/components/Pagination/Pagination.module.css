.list {
    display: flex;
    list-style: none;
}

.listButton {
    width: 40px;
    height: 40px;
    background-color: #fff;
    color: #007bff;
    border: 1px solid #eee;
    border-radius: 2px;
    outline: none;
}

.listButton:hover {
    background-color: #f0f0f0;
}

.listButton:active {
    background-color: #f0f0f0;
}

.listButton:focus {
    border: 1px solid #007bff;
}

.isActive {
    background-color: #007bff;
    color: #fff;
}

.isActive:hover {
    background-color: #007bff;
}





