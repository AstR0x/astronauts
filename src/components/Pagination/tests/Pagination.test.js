import React from 'react';
import renderer from 'react-test-renderer';

import { Pagination } from 'components/Pagination';

describe('component Pagination should', () => {
  const component = renderer.create(
    <Pagination onChangePage={() => ({})} currentPage={2} totalPages={10} />,
  );

  const tree = component.toJSON();

  it('correct render', () => {
    expect(tree).toMatchSnapshot();
  });
});
