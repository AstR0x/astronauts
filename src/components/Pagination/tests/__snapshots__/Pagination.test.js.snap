// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Pagination should correct render 1`] = `
<ul
  className="list"
>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
       &lt;&lt;
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </li>
  <li>
    <button
      className="listButton isActive"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      10
    </button>
  </li>
  <li>
    <button
      className="listButton"
      onClick={[Function]}
      type="button"
    >
      &gt;&gt;
    </button>
  </li>
</ul>
`;
